### DISCUSSION TITLE
string/constant-pool

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#102] [Announcements]
# string/constant-pool

Java程序员进阶之路，小白的零基础Java教程，从入门到进阶，深入理解Java字符串常量池

https://tobebetterjavaer.com/string/constant-pool.html

**Comment by wylgithub** [comment]
二哥请问“三妹”是真实存在的吗？还是你自己杜撰的？仅仅好奇！！！

**Reply by itwanger** [reply]
小说源于生活，又不局限于生活（😂）

**Comment by HawkSweet** [comment]
你好，请问字符串常量池放到堆中是java8开始的还是java7？ 
本节“深入理解”Java字符串常量池“里有张配图说的是java8，然而，小节”深入解析String.intern()方法“中提到”不过，需要注意的是，Java 7 的时候，字符串常量池从永久代中移动到了堆中，虽然此时永久代还没有完全被移除。Java 8 的时候，永久代被彻底移除。”，说的是java7。

**Reply by itwanger** [reply]
你的问题让我重新思考了一下，随后会调整一下文章中的示例图。

并加一些详细的说明。

**Comment by HawkSweet** [comment]
您发的邮件已经收到！谢谢~~~

**Comment by ustcjsj** [comment]
个人体会是，不如看看这个···
https://www.cnblogs.com/Andya/p/14067618.html

**Comment by HawkSweet** [comment]
您发的邮件已经收到！谢谢~~~
